cmake/modules/AddSwift.cmake

```diff
@@ -997,7 +1021,7 @@ function(_add_swift_library_single target name)
     set_target_properties("${target}"
       PROPERTIES
       INSTALL_NAME_DIR "${install_name_dir}")
-  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" AND NOT "${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
```

Comment:
It's up to app developer how it will be loaded. Between other solutions one of the popular approaches it to link everything statically into `libmyapp.so` and then load it once with `System.loadLibrary("myapp")`.

Usually you can't invoke something like `main()` in native code in Android directly (there are exceptions which usually depend on rooting your phone). Then your app starts it always starts in Java code, which then can instantly load native code and call C/Swift function of interest. Developers who don't want to use any Java code for their app can use [`NativeActivity`](http://developer.android.com/intl/ru/reference/android/app/NativeActivity.html) as a shortcut, which comes from Android SDK and does exactly this - loads your library of choice and calls predefined function `android_main`.